.github/workflows/template-notebook-test/action.yaml

```diff
@@ -0,0 +1,70 @@
+name: Notebook test template
+description: A composite action to setup and run example notebooks using Papermill
+
+inputs:
+  kubernetes-version:
+    required: true
+    description: kubernetes version
+  python-version:
+    required: false
+    description: Python version
+    # Most latest supporting version
+    default: "3.10"
+  papermill-args-yaml:
+    description: 'Additional arguments to pass to Papermill in yaml format'
+    required: false
+    default: ""
+  notebook-input:
+    description: 'Path to the input notebook'
+    required: true
+  notebook-output:
+    description: 'Path to save the output notebook'
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Free-Up Disk Space
+      uses: ./.github/workflows/free-up-disk-space
+
+    - name: Setup Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ inputs.python-version }}
+
+    - name: Install Python dependencies
+      shell: bash
+      run: |
+        pip install jupyter ipykernel papermill==2.2.0
+
+    - name: Create k8s Kind Cluster
+      uses: helm/kind-action@v1.10.0
+      with:
+        node_image: kindest/node:${{ inputs.kubernetes-version }}
+        cluster_name: training-operator-cluster
+        kubectl_version: ${{ inputs.kubernetes-version }}
+
+    - name: Build training-operator
+      shell: bash
+      run: |
+        ./scripts/gha/build-image.sh
+      env:
+        TRAINING_CI_IMAGE: kubeflowtraining/training-operator:test
+
+    - name: Deploy training operator
+      shell: bash
+      run: |
+        ./scripts/gha/setup-training-operator.sh
+        docker system prune -a -f
+        docker system df
+        df -h
+      env:
+        KIND_CLUSTER: training-operator-cluster
+        TRAINING_CI_IMAGE: kubeflowtraining/training-operator:test
+        GANG_SCHEDULER_NAME: "none"
+        KUBERNETES_VERSION: ${{ inputs.kubernetes-version }}
+
+    - name: Run Jupyter Notebook with Papermill
+      shell: bash
+      run: |
+        papermill ${{ inputs.notebook-input }} ${{ inputs.notebook-output }} -p kubeflow_python_sdk "./sdk/python" --parameters_yaml "${{ inputs.papermill-args-yaml }}"
```

Comment:
Should we create a bash script that execute our Notebooks ? In that case, you don't need to have this template action, but rather just have another GitHub action that runs this bash script.
I think that would be useful for folks, who want to tests it locally in their environment.
I was thinking that for V2, we can put similar script to execute example Notebooks under `/test/e2e/notebooks`.

Reply:
WDYT @saileshd1402 @Electronic-Waste @kubeflow/wg-training-leads @saileshd1402 @seanlaii @varshaprasad96?

Reply:
cc @akshaychitneni

Reply:
SGTM

Reply:
LGTM!

Reply:
I wanted to check which steps we should run in the bash script and which ones in github actions. For example: creating kind cluster, building, deploying on cluster, etc is specific to CI. I think it would be a good idea to keep all the setup steps in github actions itself and the script can install papermill and run the notebooks. 

We can also make a "Setup E2E" github action template with steps from [integration-tests.yaml](https://github.com/kubeflow/training-operator/blob/cb92b36bfe9ae1923a1a19e1291a9bc0cb5e181d/.github/workflows/integration-tests.yaml#L58-L94) so that we can use it for notebooks as well

Reply:
Okay yes, just wanted to make sure. thanks!

Reply:
>creating kind cluster, building, deploying training operator, etc is specific to CI. I think it would be a good idea to keep all the setup steps in github actions itself and the script can install papermill and run the notebooks.



Yes, I agree with you @saileshd1402. Let's use the GitHub action to configure cluster and deploy Training Operator.
We should re-use the same template as we use for our e2e tests.